src/main/java/de/tum/in/www1/artemis/domain/Exercise.java

```diff
@@ -385,6 +385,17 @@ else if (latestResult.getCompletionDate().isBefore(result.getCompletionDate()))
         return latestResult;
     }
 
+    /**
+     * Returns all results of an exercise for give participation
+     * (relevancy depends on Exercise type => this should be overridden by subclasses if necessary)
+     *
+     * @param participation the participation whose results we are considering
+     * @return all results of given participation, or null, if none exist
+     */
+    public Set<Result> findResultsWithCompletionDate(Participation participation) {
+        return participation.getResults();
```

Comment:
You do not filter the results for completion date here. What happens in the case results is not initialized? Even if subclasses override this method, the code should make sense and you could actually filter for results where completion date is not null

Reply:
I renamed the function to reflect the behavior better. Here I need all results for the result history in the exercise details page. But in the case of quiz exercises, a filter is required. Something like this does not exist for other types. I hope my new comment makes it more clear

Reply:
👍